lib/forums.js

```diff
@@ -713,23 +723,27 @@ var library = {
         sql_match:  (row) => {
             return mysql.format(
                 'SELECT c.id AS course, ' +
-                '       u1.id AS author_userid, u1.username AS auname, ' +
-                '       u2.id AS target_userid, u2.username AS tguname ' +
+                '       u1.id AS author_userid, u1.username AS auname, u1.email AS aumail, ' +
+                '       u2.id AS target_userid, u2.username AS tguname, u2.email AS tgmail ' +
                 'FROM mdl_course c ' +
-                'JOIN mdl_user u1 ON u1.username = ?  ' +
-                'JOIN mdl_user u2 ON u2.username = ?  ' +
+                'JOIN mdl_user u1 ON ( u1.username = ? OR u1.email = ? ) ' +
+                'JOIN mdl_user u2 ON ( u2.username = ? OR u2.email = ? )  ' +
                 'WHERE c.shortname = ?',
                 [
                     row["author_username"],
+                    row["author_email"],
                     row["target_username"],
+                    row["target_email"],
                     row["course_shortname"]
                 ]
             );
         },
 
         fixer: function(log_row, old_matches, new_matches){        
             return fix_by_match_index(log_row, old_matches, new_matches, (lr, nm) => {
-                return (lr.author_username === nm.auname) && (lr.target_username === nm.tguname);
+                return (lr.username === nm.uname || lr.email === nm.email);
+
+                return (lr.author_username === nm.auname || lr.author_email === nm.aumail) && (lr.target_username === nm.tguname || lr.target_email === nm.tgmail);
```

Comment:
This is dead code, but it looks more interesting than the first statement, so this is the one I preserved.